Scheduler documentation

AntonNiko opened the pull request:
Refactor the scheduler's documentation

Gregory-OHagan committed:
Added simple consumers and producers for tutorials

Gregory-OHagan committed:
set up tutorial start file for tutorial part 1

Gregory-OHagan committed:
slight change to demo logic

AntonNiko committed:
Added brief explanation and sequence diagram

AntonNiko committed:
Modified images

AntonNiko committed:
Modified explanation

AntonNiko committed:
Explanation changes

AntonNiko committed:
Minor changes

Commit:
Simplify and remove old readme documentation
* Move info about ROS to bottom of readme because it is less important
* Add a brief overview of the Scheduler to the top of the readme
* Remove improvements sections because the mention of Arduino/Pi communication is unnecessary
* Fix minor spelling error

AntonNiko committed:
Added docs incl. usage examples

AntonNiko committed:
Resolve merge conflict for Readme

AntonNiko committed:
Merge branch 'master' into scheduler_documentation

s-sparrow reviewed (changes requested):
It looks like sample_material.py and scheduler_start_tutorial.py are getting included in this pull request (since they were already in the branch). I would like those changes to be reverted, since they already exist in the scheduler_tutorial branch, and I don't want to clutter the master branch.

s-sparrow committed:
Add rover diagram for wiki

Add rover diagram to be used with the Scheduler wiki. We may end up moving images to a permanent branch or a separate repo.

s-sparrow committed:
Add scheduler information flow diagram

This will be linked to in the wiki and possibly the scheduler documentation.

Commit:
Remove tutorial files

These files are already in the schedule_tutorial branch which we are keeping separate for now, so I am removing them from this branch before merging the documentation
